### ISSUE TITLE
Object literal may only specify known properties, and 'authChecker' does not exist in type 'BuildSchemaOptions'.

### ISSUE THREAD

**Issue by diegonc** [MichalLytek/type-graphql#653]
**Describe the Bug**
When trying to activate an authChecker the buildSchema function throws an error.

**To Reproduce**
You can find a test repository here: https://github.com/diegonc/test-authChecker

**Expected Behavior**
buildSchema should accept the authChecker parameter

**Logs**

```
$ npm start

> test-authchecker@1.0.0 start /home/diegonc/dev/test/test-authChecker
> ts-node index.ts


/usr/lib/node_modules/ts-node/src/index.ts:240
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
index.ts:8:3 - error TS2345: Argument of type '{ authChecker: () => boolean; resolvers: [typeof TestResolver]; }' is not assignable to parameter of type 'BuildSchemaOptions'.
  Object literal may only specify known properties, and 'authChecker' does not exist in type 'BuildSchemaOptions'.

8   authChecker: () => true,
    ~~~~~~~~~~~~~~~~~~~~~~~

    at createTSError (/usr/lib/node_modules/ts-node/src/index.ts:240:12)
    at reportTSError (/usr/lib/node_modules/ts-node/src/index.ts:244:19)
    at getOutput (/usr/lib/node_modules/ts-node/src/index.ts:360:34)
    at Object.compile (/usr/lib/node_modules/ts-node/src/index.ts:393:11)
    at Module.m._compile (/usr/lib/node_modules/ts-node/src/index.ts:439:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/usr/lib/node_modules/ts-node/src/index.ts:442:12)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! test-authchecker@1.0.0 start: `ts-node index.ts`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the test-authchecker@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/diegonc/.npm/_logs/2020-06-16T16_38_43_363Z-debug.log
```

**Environment (please complete the following information):**

- OS: Kubuntu 19.10 (eoan)
- Node: v12.16.3
- Package version: 1.0.0-rc.2 (as installed by `npm install --save type-graphql`)
- TypeScript version:
    3.4.5 (according to `/usr/lib/node_modules/typescript/package.json` )
     v8.1.0 (ts-node -v)

**Comment by MichalLytek** [comment]
Can you try upgrading the TS version?

**Comment by MichalLytek** [comment]
And read the installation docs because your tsconfig is wrong

**Comment by diegonc** [comment]
I followed the installation docs and put the minimal config in tsconfig. Also I installed the missing packages and imported `reflect-metadata`.
I also did `npm i -g typescript` which installed version 3.9.5
Everything was committed to the test repo. But the error persisted.

Finally, I ran `npm i -g ts-node` which installed version v8.10.2 and fixed the error.

Thanks for helping troubleshoot this exception!
